vault/resource_quota_rate_limit_test.go

```diff
@@ -68,6 +68,37 @@ func TestQuotaRateLimit(t *testing.T) {
 	})
 }
 
+func TestQuotaRateLimitWithRole(t *testing.T) {
+	name := acctest.RandomWithPrefix("rate-limit")
+	backend := acctest.RandomWithPrefix("approle")
+	role := acctest.RandomWithPrefix("test-role")
+	rateLimit := randomQuotaRateString()
+	resourceName := "vault_quota_rate_limit.foobar"
+
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			testQuotaRateLimitCheckDestroy([]string{rateLimit}),
+			testAccCheckAppRoleAuthBackendRoleDestroy,
+		),
+		Steps: []resource.TestStep{
+			{
+				Config: testQuotaRateLimitWithRoleConfig(backend, role, name, rateLimit, 1, 0),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, fmt.Sprintf("auth/%s/", backend)),
+					resource.TestCheckResourceAttr(resourceName, "rate", rateLimit),
+					resource.TestCheckResourceAttr(resourceName, "interval", "1"),
+					resource.TestCheckResourceAttr(resourceName, "block_interval", "0"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldRole, role),
+				),
+			},
+			testutil.GetImportTestStep(resourceName, false, nil, "name", consts.FieldRole),
```

Comment:
I am not sure why the role is not getting imported in the test. I see the role when I do the manual import

```shell
$ terraform import vault_quota_rate_limit.foobar transit-limit
vault_quota_rate_limit.foobar: Importing from ID "transit-limit"...
vault_quota_rate_limit.foobar: Import prepared!
  Prepared vault_quota_rate_limit for import
vault_quota_rate_limit.foobar: Refreshing state... [id=transit-limit]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.

$ terraform state show vault_quota_rate_limit.foobar
# vault_quota_rate_limit.foobar:
resource "vault_quota_rate_limit" "foobar" {
    block_interval = 0
    id             = "transit-limit"
    interval       = 60
    path           = "auth/approle/"
    rate           = 1000
    role           = "my-role"
}
```

Reply:
cc @vinay-gopalan Any idea why this is happening in the test?

Reply:
I just ran both tests locally without including the `role` field in the ignore list, and they both seem to pass. Looking at the [Vault docs](https://developer.hashicorp.com/vault/api-docs/system/rate-limit-quotas#sample-response) for the feature as well, it looks like the `role` field will always be returned on a read for both the resources, so I think we should be good to remove it from the ignore lists 👍🏼 

```suggestion
			testutil.GetImportTestStep(resourceName, false, nil, "name"),
```

Reply:
Thanks for the double check! I think the test failed the first time I ran it without including the role, so I assumed that I had done something wrong and included it in the list 🤔. I'll remove it.

Reply:
@fairclothjm @vinay-gopalan Note that CI just failed after I removed the "role" attribute from this list. Is there something else I need to do?

Reply:
Ah I see what's going on. We can see CI failing on 1.11, and that's likely because we're not using the version check for this field. I think this should be fixed if we only write/read this to and from Vault if it satisfies the minimum version requirement (minimum version required in this case would be 1.12, since the role field does not exist on 1.11).

My understanding is 1.11 is failing because we are attempting to always write the `role` field to Vault, and that is different from what we're getting back from Vault (no `role` field will be returned on 1.11). I believe this can be resolved by adding the version check as done [here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_aws_auth_backend_client.go#L124). I'll also try and add a comment suggestion to give more context

Reply:
@vinay-gopalan Ah ha! Thank you for that insight. I'll check out the link you added.

Reply:
Ah, we need to add `name` to the `quotaRateLimitRead` fields that we then `d.Set(...)`. That should allow us to import both `role` and `name` which are both returned by the API

Reply:
@fairclothjm Thanks for the pointer. I added that here https://github.com/hashicorp/terraform-provider-vault/pull/1994/commits/9ff1f50adf9dae1d239d973a5fdcfc2bf1ada3a6